project/src/com/google/daggerquery/executor/BUILD

```diff
@@ -24,4 +24,7 @@ java_binary(
         "//src/com/google/daggerquery/executor/models:query_executor_models",
         "//src/com/google/daggerquery/executor/services:query_executor_services",
     ],
+    resources = [
+        "//src/com/google/daggerquery:binding_graph_data"
```

Comment:
I'm not sure it makes sense to hard code the textproto to query_executor. How about creating a macro (like the `dagger_query_textproto`) so that this dependency can be specified by the user via arguments?

Reply:
Yes, it might make sense not to hardcode it here. However I think I didn't understand how this macro will be used - you mean user don't need to specify anything in `resources` at all by himself?

Reply:
I was thinking if the user could just call your macro from their project's `BUILD` file, e.g.

```
# user's BUILD file
dagger_query(
  name = "app_dagger_query",
  dagger_app_target = "MY_APP_TARGET"
)
```

and then in your implementation:

```
# your macro implementation that accepts user arguments
def dagger_query(name, dagger_app_target):
   dagger_query_textproto(
         name = "binding_graph_data",
         dagger_app_target = "YOUR_TARGET"
      )

     java_binary(
       name = name
       deps = ["//src/com/google/daggerquery/executor:query_executor"],
       resources = [":binding_graph_data"],
     )
```

and your `dagger-query.sh` script may need to take `//...:app_dagger_query` as a user supplied argument as well.

Reply:
(The question was discussed below, resolving this conversation as all changes made in the latest commits.) ✅